app/services/studentLoans/StudentLoansService.scala

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services.studentLoans
+
+import models.User
+import models.employment.StudentLoansCYAModel
+import models.mongo.{EmploymentCYAModel, EmploymentUserData}
+import services.EmploymentSessionService
+import utils.Clock
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class StudentLoansService @Inject()(employmentSessionService: EmploymentSessionService,
+                                    implicit val ec: ExecutionContext) {
+
+  def updateUglQuestion(taxYear: Int, employmentId: String, originalEmploymentUserData: EmploymentUserData, ugl: Boolean)
+                           (implicit user: User[_], clock: Clock): Future[Either[Unit, EmploymentUserData]] = {
+    val cya = originalEmploymentUserData.employment
+    val newStudentLoans: Option[StudentLoansCYAModel] = originalEmploymentUserData.employment.studentLoansCYAModel.
+      map(studentLoan => if (ugl) {
+        studentLoan.copy(uglDeduction = ugl, uglDeductionAmount = None)
+      } else {
+        studentLoan.copy(uglDeduction = ugl, uglDeductionAmount = None)
+      })
+    val updatedEmployment: EmploymentCYAModel = cya.copy(studentLoansCYAModel = newStudentLoans)
+
+    employmentSessionService.createOrUpdateEmploymentUserDataWith(
+      taxYear,
+      employmentId,
+      originalEmploymentUserData.isPriorSubmission,
+      originalEmploymentUserData.hasPriorBenefits,
+      updatedEmployment
+    )
+  }
+  def updatePglQuestion(taxYear: Int, employmentId: String, originalEmploymentUserData: EmploymentUserData, pgl: Boolean)
+                           (implicit user: User[_], clock: Clock): Future[Either[Unit, EmploymentUserData]] = {
+    val cya = originalEmploymentUserData.employment
+    val newStudentLoans: Option[StudentLoansCYAModel] = originalEmploymentUserData.employment.studentLoansCYAModel.
+      map(studentLoan => if (pgl) {
+        studentLoan.copy(pglDeduction = pgl, pglDeductionAmount = None)
+      } else {
+        studentLoan.copy(pglDeduction = pgl, pglDeductionAmount = None)
+      })
```

Comment:
The pglDeductionAmount is being wiped here no matter whether the question was updated to yes or no. This seems wrong